### PR TITLE
feat(oidc-provider-nest): NodeMailer can now use Gmail to send emails

### PR DESCRIPTION
**Pull request by atharmon**
Requires a `mailerconfig.ts` file in the environments directory with an exported member like:
`export const mailerConfig = {
  user: string (email you've setup Google API for,
  accessToken: string,
  clientId: string,
  clientSecret: string,
  refreshToken: string
};`
You'll need to go through some steps to setup `mailerConfig` for your own email or I can show you it in action with mine.

### CODE REVIEW

File: apps/oidc-provider-nest/src/main.ts [hunk 1]
```diff
@@ -103,6 +112,11 @@ async function bootstrap() {
 
   const dir = join(__dirname, 'assets/views');
 
+  // This will setup the Mailer (gmail or ethereal)
+  // Mailer.build('gmail', mailerConfig);
+  Mailer.build('ethereal');
+  Mailer.sendPasswordResetConfirmationEmail('atharmon@tamu.edu');
```
Comment (Proantagonist) [inline apps/oidc-provider-nest/src/main.ts:118]: What's up with this hard-coded email?

File: libs/oidc/common/src/lib/utils/email/mailer.util.ts [hunk 2]
```diff
@@ -3,18 +3,59 @@ import Mail from 'nodemailer/lib/mailer';
...
+      case 'ethereal':
+        // use ethereal email and print out links to console (debugging / dev)
+        Mailer.transporter = nodemailer.createTransport({
+          host: 'smtp.ethereal.email',
+          port: 587,
+          secure: false,
+          auth: {
+            user: 'kaitlyn.schimmel@ethereal.email',
+            pass: 'Pe6D9DhkgUDyqyBMeg'
+          }
```
Comment (Proantagonist) [inline libs/oidc/common/src/lib/utils/email/mailer.util.ts:35]: I can only assume this shouldn't be here?

**Review by Proantagonist — CHANGES_REQUESTED** [review]
Gotta fix some hard-coded configs

File: libs/oidc/common/src/lib/utils/email/mailer.util.ts (hunk 2 above)
Comment (atharmon) [reply libs/oidc/common/src/lib/utils/email/mailer.util.ts:35]: That depends. It's a free account that was automatically generated. Should it be there? No. Does it really matter? Not particularly.

File: apps/oidc-provider-nest/src/main.ts (hunk 1 above)
Comment (atharmon) [reply apps/oidc-provider-nest/src/main.ts]: Was probably testing the functionality of the different mailing options

File: libs/oidc/common/src/lib/utils/email/mailer.util.ts (hunk 2 above)
Comment (Proantagonist) [reply libs/oidc/common/src/lib/utils/email/mailer.util.ts:35]: image

**Review by Proantagonist — APPROVED** [review]

**Pull request merged**